pkg/appmanager/appManager.go

```diff
@@ -944,14 +960,95 @@ func (appMgr *Manager) syncIngresses(
 					appMgr.recordIngressEvent(ing, "ResourceConfigured", msg, "")
 				}
 			}
-
 			// Set the Ingress Status IP address
 			appMgr.setIngressStatus(ing, rsCfg)
 		}
 	}
 	return nil
 }
 
+func (appMgr *Manager) syncRoutes(
+	stats *vsSyncStats,
+	sKey serviceQueueKey,
+	rsMap ResourceMap,
+	svcPortMap map[int32]bool,
+	svc *v1.Service,
+	appInf *appInformer,
+) error {
+	routeByIndex, err := appInf.routeInformer.GetIndexer().ByIndex(
+		"namespace", sKey.Namespace)
+	if nil != err {
+		log.Warningf("Unable to list routes for namespace '%v': %v",
+			sKey.Namespace, err)
+		return err
+	}
+
+	for _, obj := range routeByIndex {
+		// We need to look at all routes in the store, parse the data blob,
+		// and see if it belongs to the service that has changed.
+		route := obj.(*routeapi.Route)
+		if route.ObjectMeta.Namespace != sKey.Namespace {
+			continue
+		}
+
+		rsCfg, err := createRSConfigFromRoute(route, *appMgr.resources, appMgr.routeConfig)
+		if err != nil {
+			// We return err if there was an error creating a rule
+			log.Warningf("%v", err)
+			continue
+		}
+
+		rsName := rsCfg.Virtual.VirtualServerName
+		if ok, found, updated := appMgr.handleConfigForType(
+			&rsCfg, sKey, rsMap, rsName, svcPortMap, svc, appInf,
+			route.Spec.To.Name); !ok {
+			stats.vsUpdated += updated
+			continue
+		} else {
+			stats.vsFound += found
+			stats.vsUpdated += updated
+		}
+
+		// We store this same config on every route that has the same protocol, but it is only
+		// written to the BIG-IP once. This block guarantees that all of these configs
+		// are in the same state.
+		appMgr.resources.Lock()
+		cfgs, keys := appMgr.resources.GetAllWithName(rsCfg.Virtual.VirtualServerName)
+		for i, cfg := range cfgs {
+			if cfg.Virtual.Partition == rsCfg.Virtual.Partition &&
+				!reflect.DeepEqual(cfg, rsCfg) {
+				cfg = &rsCfg
+				appMgr.resources.Assign(keys[i], cfg.Virtual.VirtualServerName, cfg)
+			}
+		}
+		appMgr.resources.Unlock()
+
+		// TLS Cert/Key
+		if nil != route.Spec.TLS && len(route.Spec.TLS.Termination) > 0 {
```

Comment:
Should probably check for Termination == "edge" here rather than the length chck, as the logic will be different for the other values ("passthrough" and "reencrypt").